emily/lambda/src/errors.rs

```diff
@@ -0,0 +1,152 @@
+            // EmilyApiError::Conflict(_) => http::StatusCode::CONFLICT,
+            // EmilyApiError::NotImplemented(_) => http::StatusCode::NOT_IMPLEMENTED,
+            // EmilyApiError::Throttling(_) => http::StatusCode::TOO_MANY_REQUESTS,
+            // EmilyApiError::InternalService(_) => http::StatusCode::INTERNAL_SERVER_ERROR,
+            EmilyApiError::UnhandledService(_) => http::StatusCode::INTERNAL_SERVER_ERROR,
+        }.as_u16()
+    }
+
+    fn response_body(&self) -> Result<String, serde_json::Error> {
+        match *self {
+            EmilyApiError::BadRequest(_) =>
+                serde_json::to_string(&models::BadRequestErrorResponseContent { message: self.to_string() }),
+            // EmilyApiError::Forbidden(_) =>
+            //     serde_json::to_string(&ForbiddenErrorResponseContent { message: self.to_string() }),
+            // EmilyApiError::NotFound(_) =>
+            //     serde_json::to_string(&NotFoundErrorResponseContent { message: self.to_string() }),
+            // EmilyApiError::Conflict(_) =>
+            //     serde_json::to_string(&ConflictErrorResponseContent { message: self.to_string() }),
+            // EmilyApiError::NotImplemented(_) =>
+            //     serde_json::to_string(&NotImplementedErrorResponseContent { message: self.to_string() }),
+            // EmilyApiError::Throttling(_) =>
+            //     serde_json::to_string(&ThrottlingErrorResponseContent { message: self.to_string() }),
+            // EmilyApiError::InternalService(_) =>
+            //     serde_json::to_string(&models::ServiceErrorResponseContent { message: self.to_string() }),
+            EmilyApiError::UnhandledService(_) =>
+                serde_json::to_string(&models::ServiceErrorResponseContent { message: self.to_string() }),
+        }
+    }
+
+    #[allow(clippy::wrong_self_convention)]
+    pub fn to_apigw_response(self) -> ApiGatewayProxyResponse {
+
+        let status_code = self.status_code();
+        let body_result = self.response_body();
+
+        match body_result {
+            Ok(body) => common::SimpleApiResponse {
+                status_code: status_code,
+                body: Some(Body::Text(body)),
+            },
+            // This occurs in the rare case that the API Error itself failed to serialize.
+            Err(err) => common::SimpleApiResponse {
+                status_code: http::StatusCode::INTERNAL_SERVER_ERROR.as_u16(),
+                // This output won't be of the right format for the client, but this is an edge case where
+                // the serialization will have needed to fail on a well defined structure, and this error
+                // can be handled by someone receiving data from the client since they'll need to handle
+                // potential errors anyway.
+                body: Some(Body::Text(serde_json::json!({"message": format!("Server error {err} on {self}")}).to_string()))
```

Comment:
Yeah I think last time I had `{err}` in the string, but I now think that this would be a slight mistake. We shouldn't display our internal error in the response.